mongo/datadog_checks/mongo/mongo.py

```diff
@@ -335,206 +219,22 @@ def _authenticate(self, database):
-                    ]
-                    val = int(stats.get('accesses', {}).get('ops', 0))
-                    self.gauge('mongodb.collection.indexes.accesses.ops', val, idx_tags)
-            except Exception as e:
-                self.log.error("Could not fetch indexes stats for collection %s: %s", coll_name, e)
-
-    def _get_submission_method(self, method_name):
-        if method_name not in ALLOWED_CUSTOM_METRICS_TYPES:
-            raise ValueError('Metric type {} is not one of {}.'.format(method_name, ALLOWED_CUSTOM_METRICS_TYPES))
-        return getattr(self, method_name)
-
-    @staticmethod
-    def _extract_command_from_mongo_query(mongo_query):
-        """Extract the command (find, count or aggregate) from the query. Even though mongo and pymongo are supposed
-        to work with the query as a single document, pymongo expects the command to be the `first` element of the
-        query dict.
-        Because python 2 dicts are not ordered, the command is extracted to be later run as the first argument
-        of pymongo `runcommand`
-        """
-        for command in ALLOWED_CUSTOM_QUERIES_COMMANDS:
-            if command in mongo_query:
-                return command
-        raise ValueError("Custom query command must be of type {}".format(ALLOWED_CUSTOM_QUERIES_COMMANDS))
-
-    def _collect_custom_metrics_for_query(self, db, raw_query, tags):
-        """Validates the raw_query object, executes the mongo query then submits the metrics to datadog"""
-        metric_prefix = raw_query.get('metric_prefix')
-        if not metric_prefix:  # no cov
-            raise ValueError("Custom query field `metric_prefix` is required")
-        metric_prefix = metric_prefix.rstrip('.')
-
-        mongo_query = deepcopy(raw_query.get('query'))
-        if not mongo_query:  # no cov
-            raise ValueError("Custom query field `query` is required")
-        mongo_command = self._extract_command_from_mongo_query(mongo_query)
-        collection_name = mongo_query[mongo_command]
-        del mongo_query[mongo_command]
-        if mongo_command not in ALLOWED_CUSTOM_QUERIES_COMMANDS:
-            raise ValueError("Custom query command must be of type {}".format(ALLOWED_CUSTOM_QUERIES_COMMANDS))
-
-        submit_method = self.gauge
-        fields = []
-
-        if mongo_command == 'count':
-            count_type = raw_query.get('count_type')
-            if not count_type:  # no cov
-                raise ValueError('Custom query field `count_type` is required with a `count` query')
-            submit_method = self._get_submission_method(count_type)
+    def update_deployment_type(self, admindb):
+        props = admindb.command("isMaster")
+        if props.get("ismaster") == "isdbgrid":
+            self.deployment_type = MongosDeploymentType()
+        elif props.get("hosts"):
+            repl_set_payload = admindb.command("replSetGetStatus")
+            self.deployment_type = ReplicaSetDeploymentType(repl_set_payload)
```

Comment:
Related to my comment on `ReplicaSetDeploymentType`, here we should do:

```suggestion
            repl_set_payload = admindb.command("replSetGetStatus")
            replset_name = repl_set_payload["name"]
            replset_state = repl_set_payload["myState"]
            self.deployment_type = ReplicaSetDeploymentType(replset_name, replset_state)
```